vyper/context/types/function.py

```diff
@@ -376,23 +376,13 @@ def from_AnnAssign(cls, node: vy_ast.AnnAssign) -> "ContractFunction":
         )
 
     @property
-    def method_ids(self) -> Dict[str, int]:
+    def method_id(self) -> int:
         """
-        Dict of `{signature: four byte selector}` for this function.
-
-        * For functions without default arguments the dict contains one item.
-        * For functions with default arguments, there is one key for each
-          function signfature.
+        The primary method ID, used to represent this function internally
+        regardless of the number of default arguments given.
```

Comment:
I think this needs more thought.

The "default" should be chosen probably based on how it's best represented, and explain why this is the case.